framework/src/st_things/things_stack/framework/things_req_handler.c

```diff
@@ -271,12 +274,15 @@ static OCEntityHandlerResult get_provisioning_info(things_resource_s *target_res
 {
 	OCEntityHandlerResult eh_result = OC_EH_ERROR;
 
-	const char *device_id = OCGetServerInstanceIDString();
 	const char *device_rt = dm_get_things_device_type(0);
 	bool is_owned = false;
 	bool is_reset = things_get_reset_mask(RST_ALL_FLAG);
 
-	if (OC_STACK_OK != OCGetDeviceOwnedState(&is_owned)) {
+	iotivity_api_lock();
+	const char *device_id = OCGetServerInstanceIDString();
```

Comment:
Please note that device_id points to static string (static char [37]) allocated by iotivity stack. While its lifetime is till program's termination, use it for caution (especially during termination of stack as exact point when it will be destroyed during program termination is not defined).

Reply:
Ok @abhishek16j.